src/main/java/com/cloudbees/hudson/plugins/folder/computed/OrphanedParent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2022 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.hudson.plugins.folder.computed;
+
+import hudson.model.Item;
+import jenkins.model.CauseOfInterruption;
+
+/**
+ * Indicates that an execution was aborted because of the orphaning of one of its parents.
+ *
+ * @author Réda Housni Alaoui
+ */
+public class OrphanedParent extends CauseOfInterruption {
+
+	private final String orphanedParentFullName;
+
+	public OrphanedParent(Item orphanedParent) {
+		this.orphanedParentFullName = orphanedParent.getFullName();
```

Comment:
```suggestion
		this.orphanedParentFullName = orphanedParent.getFullDisplayName();
```

(if indeed used only for display)